File: others/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
     hash := make(map[int]int)
     ret := make([]int, 0)
     for i := 0; i < len(nums); i++ {
        hash[nums[i]] = i
     }
     for i := 0; i < len(nums); i++ {
         ind, exists := hash[target-nums[i]]
         if exists && i!=ind {
            ret = append(ret,ind)
            ret = append(ret,i)
            break
         }
     }

    return ret
}

func main(){
    arg := []int{2,7,11,15}
    ret := twoSum(arg,9)
    fmt.Println(ret)
    arg = []int{3,2,4}
    ret = twoSum(arg,6)
    fmt.Println(ret)
}

File: others/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let tmp=[]
    nums.forEach((val,ind) => tmp.push([val, ind]));
    nums = tmp;
    nums.sort((a,b) => a[0]-b[0]);
    let i=0,j=nums.length-1,ret=[];
    while(i<j){
        let tmp=nums[i][0]+nums[j][0];
        if(tmp==target){
            ret.push(nums[i][1]);
            ret.push(nums[j][1]);
            i++;
            j--;
        } else if(tmp<target){
            i++;
        } else{
            j--;
        }
    }
    ret.sort((a,b) => a-b);
    return ret;
};
console.log(twoSum([2,7,11,15],9));
console.log(twoSum([3,2,4],6));
